### ISSUE TITLE
Overloads for `Literal[True/False]` and `bool` parameters marked as overlapping

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#3408]
**Describe the bug**
Given the following code:

```python
from __future__ import annotations

from typing import Sequence, TypeVar, overload
from typing_extensions import Literal

_Record = TypeVar('_Record')


@overload
def foo(s: Sequence[_Record], extra: Literal[False]) -> list[_Record]:
    ...


@overload
def foo(
    s: Sequence[_Record], extra: Literal[True]
) -> tuple[list[_Record], bytes, bool]:
    ...


@overload
def foo(
    s: Sequence[_Record], extra: bool
) -> list[_Record] | tuple[list[_Record], bytes, bool]:
    ...


def foo(
    s: Sequence[_Record], extra: bool
) -> list[_Record] | tuple[list[_Record], bytes, bool]:
    result = list(s)
    if extra:
        return (result, b'', True)
    else:
        return result


def baz(bar: bool) -> None:
    f1 = foo([1, 2, 3], False)
    f2 = foo([1, 2, 3], True)
    f3 = foo([1, 2, 3], bar)
    reveal_type(f1)
    reveal_type(f2)
    reveal_type(f3)
```

The overload at line 10 is marked as an error:

```
test.py:10:5 - error: Overload 1 for "foo" overlaps overload 3 and returns an incompatible type
```

Normally, I would remove the third overload, but removing it (for this example's purposes, I'm commenting it out to preserve line numbers) causes an error at line 41:

```
test.py:41:25 - error: Argument of type "bool" cannot be assigned to parameter "extra" of type "Literal[True]" in function "foo"
    "bool" cannot be assigned to type "Literal[True]"
test.py:41:5 - error: Type of "f3" is unknown
```

**Expected behavior**
No errors should be marked for line 10 (the first overload) when all three overloads are present.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.241

**Additional context**
`mypy` does not mark line 10 as an error in strict mode, but does mark line 41 as an error when the third overload is commented out. As I understand, `bool` is incompatible with `Literal[True] | Literal[False]`, so the current `pyright` behavior seems to match what is expected. This leads me to believe that line 10 should not be marked as an error.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

The problem had nothing to do with Literal[True/False]. It was a bug that affected any overlapping overloads with generic return types where the TypeVar was scoped to the function (as opposed to an enclosing class).

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.242, which I just published. It will also be included in the next release of pylance.
